functions/java-jre.bash

```diff
@@ -231,3 +261,47 @@ java_zulu_install_crypto_extension(){
   rm -rf "${policyTempLocation}"
   return 0
 }
+
+## Install AdoptOpenJDK using APT repository.
+##
+adoptopenjdk_install_apt(){
+  local adoptKey="/tmp/adoptopenjdk.asc"
+  if ! dpkg -s 'adoptopenjdk-8-hotspot-jre' >/dev/null 2>&1; then # Check if already is installed
+    echo -n "$(timestamp) [openHABian] Adding AdoptOpenJDK keys to apt... "
+    cond_redirect wget -qO "$adoptKey" https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public
+    if cond_redirect apt-key add "$adoptKey"; then
+      echo "OK"
+    else
+      echo "FAILED (keyserver)"
+      rm -f "$adoptKey"
+      exit 1;
+    fi
+    echo -n "$(timestamp) [openHABian] Installing AdoptOpenJDK... "
+    cond_redirect systemctl stop openhab2.service
+    cond_redirect update-alternatives --remove-all java
+    echo "deb http://adoptopenjdk.jfrog.io/adoptopenjdk/deb buster main" > /etc/apt/sources.list.d/adoptopenjdk.list
+    cond_redirect apt-get update
+    if cond_redirect apt-get --yes install adoptopenjdk-8-hotspot-jre; then echo "OK"; else echo "FAILED"; exit 1; fi
+    cond_redirect systemctl start openhab2.service
+  fi
+}
+
+## Uninstall AdoptOpenJDK using APT repository.
+##
+adoptopenjdk_uninstall_apt(){
+  if dpkg -s 'adoptopenjdk-8-hotspot-jre' >/dev/null 2>&1; then
+    echo -n "$(timestamp) [openHABian] Removing AdoptOpenJDK... "
+    cond_redirect update-alternatives --remove-all java
```

Comment:
to be frank I'm unaware how update-alternatives works, but wouldn't that remove references to *all* installed java packages including Zulu ?

Reply:
Yes, but then it is re-added once it is configured for Zulu, and because of the way it is chosen it will properly add them again when installing.